File: string-paradigm-no-wrapper/src/main/java/org/jesperancinha/string/paradigm/nowrapper/performance/NoWrapperDependencyImpl.java
package org.jesperancinha.string.paradigm.nowrapper.performance;

import java.util.List;

import com.jesperancinha.string.paradigm.api.Dependency;

public class NoWrapperDependencyImpl extends Dependency {
	private String dependencies;

	public NoWrapperDependencyImpl(String data) {
		this.dependencies = data;
	}

	public int getDependencyCount() {
		return dependencies.toString().split(";").length;
	}

	public String getSubDependencyElement(int index) {
		final String[] splittedValues = dependencies.toString().split(";");
		if (index < splittedValues.length - 1) {
			return splittedValues[index];
		} else {
			return null;
		}
	}

	public boolean startsWith(List<String> subDependencyElement) {
		int codeCount = getDependencyCount();
		if (codeCount < subDependencyElement.size()) {
			return false;
		}
		for (int i = 0; i < subDependencyElement.size(); i++) {
			if (!getSubDependencyElement(i).equals(subDependencyElement.get(i))) {
				return false;
			}
		}
		return true;
	}
}

File: jeorg-kotlin-apps/string-array-paradigms/string-paradigm-no-wrapper/src/main/kotlin/org/jesperancinha/string/paradigm/nowrapper/performance/NoWrapperDependencyImpl.kt
package org.jesperancinha.string.paradigm.nowrapper.performance

import org.jesperancinha.string.paradigm.api.Dependency

class NoWrapperDependencyImpl(private val dependencies: String) : Dependency() {
    override fun dependencyCount(): Int {
        return dependencies.split(";".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray().size
    }

    override fun getSubDependencyElement(index: Int): String? {
        val splittedValues = dependencies.split(";".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
        return if (index < splittedValues.size - 1) {
            splittedValues[index]
        } else {
            null
        }
    }

    override fun startsWith(subDependencyElement: List<String>): Boolean {
        val codeCount = dependencyCount()
        if (codeCount < subDependencyElement.size) {
            return false
        }
        for (i in subDependencyElement.indices) {
            if (getSubDependencyElement(i) != subDependencyElement[i]) {
                return false
            }
        }
        return true
    }
}